.github/workflows/publish-docker-image.yml

```diff
@@ -0,0 +1,35 @@
+name: Publish Docker image
+
+on:
+  push:
+    paths-ignore:
+      - README.md
+      - ansible/*
+      - .vscode/*
+      - .venv/*
+      - .pytest_cache/*
+      - '**__pycache__/*'
+      - tests/.pytest_cache/*
+    branches:
+      - master
+      - exercise-8 # TODO exercise-8: Remove once have tested - only want to create image when master updated
+
+jobs:
+  push_to_registry:
```

Comment:
Minor, but note that because you've created this as a separate workflow rather than a second job in the original file, it means that the deployment will run even if the tests fail on master (and I believe there's still no way to add a dependency on another workflow). That's valid if your processes account for it - e.g. you don't expect people to handle merges locally where mistakes might be introduced just as they push to master, or you just accept that risk and that this is a test environment and occasional issues will happen - but might not be intentional/obvious.

Reply:
Hm I've had a go at this with conditionals, so will merge using that. Feels a bit ugly though.